packages/schematics/update/update/index.ts

```diff
@@ -850,11 +873,16 @@ export default function(options: UpdateSchema): Rule {
 
   return (tree: Tree, context: SchematicContext) => {
     const logger = context.logger;
-    const allDependencies = _getAllDependencies(tree);
-    const packages = _buildPackageList(options, allDependencies, logger);
+    const allDeps = _getAllDependencies(tree);
+
+    // Filter out non-NPM dependencies by checking if they link to a semver.
+    const npmDeps = new Map(Array.from(allDeps.entries())
+        .filter(([_, source]) => isSimpleSemVer(source)));
```

Comment:
Consider using [npm-package-arg's](https://github.com/npm/npm-package-arg) `resolve` function to both validate and check for a registry specifier.  This is the logic that npm itself uses.

Reply:
Thanks for the tip, all my Google searches were leading me to the `semver` package. Updated to use `npm-package-arg` and also added a test to verify the behavior.